tests/src/testcommon.hh

```diff
@@ -187,139 +186,88 @@ template <typename NonLinearOperator>
-  /// addition of C*M*alpha to the stress from pure displacement formulation. Hence it is sufficient to test only for
-  /// one of the EAS variant as the other M matrices are already tested.
-  if constexpr (requires { fe.setEASType(4); }) {
-    fe.setEASType(4);
-    expectedStress = {{{1214.2857143755, 1214.2857143755, 384.6153846800},
-                       {1214.2857144055, 214.2857142945, 384.6153847400},
-                       {214.2857142945, 1214.2857144055, 384.6153845600},
-                       {214.2857143245, 214.2857143245, 384.6153846200}}};
-  }
-
-  auto resultRequirements = Ikarus::ResultRequirements<>()
-                                .insertGlobalSolution(Ikarus::FESolutions::displacement, displacement)
-                                .addResultRequest(ResultType::linearStress);
-
-  ResultTypeMap result;
-  auto gridView        = fe.localView().globalBasis().gridView();
-  auto scalarBasis     = makeConstSharedBasis(gridView, lagrangeDG<1>());
-  auto localScalarView = scalarBasis->localView();
-  std::vector<Dune::FieldVector<double, 3>> stressVector(scalarBasis->size());
-
-  auto ele = elements(gridView).begin();
-  localScalarView.bind(*ele);
-  const auto& fe2              = localScalarView.tree().finiteElement();
-  const auto& referenceElement = Dune::ReferenceElements<double, gridDim>::general(ele->type());
-  for (const auto c : std::views::iota(0u, fe2.size())) {
-    const auto fineKey                        = fe2.localCoefficients().localKey(c);
-    const auto nodalPositionInChildCoordinate = referenceElement.position(fineKey.subEntity(), fineKey.codim());
-    fe.calculateAt(resultRequirements, nodalPositionInChildCoordinate, result);
-    Eigen::Vector3d computedResult = result.getResult(ResultType::linearStress);
-    const auto nodeIndex           = localScalarView.index(localScalarView.tree().localIndex(c))[0];
-    stressVector[nodeIndex]        = toDune(computedResult);
-    for (const auto voigtIndex : std::views::iota(0u, static_cast<unsigned int>(gridDim * (gridDim + 1) / 2))) {
-      const auto FEStressComponent       = stressVector[nodeIndex][voigtIndex];
-      const auto expectedStressComponent = expectedStress[c][voigtIndex];
-      const bool isStressCorrect
-          = Dune::FloatCmp::eq<double, Dune::FloatCmp::CmpStyle::absolute>(FEStressComponent, expectedStressComponent);
-      t.check(isStressCorrect) << "Stress component " << voigtIndex << " at node " << c << " is not correctly computed"
-                               << messageIfFailed;
-    }
-  }
-  return t;
-}
-
-template <typename NonLinearOperator, typename FiniteElement>
-[[nodiscard]] auto checkLinearStressOfSimplex(NonLinearOperator& nonLinearOperator, FiniteElement& fe,
-                                              const std::string& messageIfFailed = "") {
+[[nodiscard]] auto checkLinearStress(NonLinearOperator& nonLinearOperator, FiniteElement& fe,
```

Comment:
Nice generalization.
I still would aim for a further abstraction level here.
I envision the signature of this function more like:

```cpp
template <typename NonLinearOperator, typename FiniteElement>
[[nodiscard]] auto checkCalculateAt(NonLinearOperator& nonLinearOperator, FiniteElement& fe,const ResultRequirements& resultRequirements,  std::vector<Eigen::Matrix...>& expectedResults,  const std::vector<LocalCoordinates>& localPositionsOfResults
                                              const std::string& messageIfFailed = "") 
```

Now we cann pass in from the outside which results we require and we can also set global solutions:

```cpp

  displacement << 0, 0, 1, 1, 1, 1, 1, 1;
  std::array<Dune::FieldVector<double, 3>, 4> expectedStress = {{{1428.5714287000, 1428.5714287000, 769.2307693000},
                                                                 {1098.9010990000, 329.6703297000, 384.6153846500},
                                                                 {329.6703297000, 1098.9010990000, 384.6153846500},
                                                                 {0, 0, 0}}};
  auto resultRequirements = ResultRequirements()
                                .insertGlobalSolution(FESolutions::displacement, displacement)
                                .addResultRequest(ResultType::linearStress);

std::array<LocalCoordinate,4> positions;
for (auto i : std::views::iota(0, nodes)) 
    positions[i] = referenceElement.position(i, dim);

checkCalculateAt(nonLinOp,fe,resultRequirements,expectedStress ,positions);
```

To really support std::array and std::vector for expected results and coordinate position in one function, you can look int [std::span<T,N](https://en.cppreference.com/w/cpp/container/span), which allows static and dynamic extend

Reply:
Excepte we refactor the calculateAt signature obviously

Reply:
Furthermore, at the outside this leads to some code duplication but i think you can write then also a wrapper.
like createLinearStressResults or something, which return you `resultRequirements,expectedStress ,positions`.

by aware you can do this via structured bindings.

```cpp
auto createLinearStressResults(...)
{
//...
return std::make_tuple(resultRequirements,expectedStress ,positions);
}
```

and at the call site this would look like

```cpp
const auto [resultRequirements,expectedStress ,positions] =createLinearStressResults(...); //directly un-pack tuple
checkCalculateAt(nonLinOp,fe,resultRequirements,expectedStress ,positions);
```

Reply:
At https://github.com/ikarus-project/ikarus/blob/7e793d0ae955b4b11ba7d2c98522775495abe703/tests/src/testlinearelasticity.cpp#L43C52-L43C52 you now have to provide your own functor.

Which can be obtained via:

```cpp

auto checkCalculateAtFactory(const auto& resultRequirements,const auto& expectedStress ,const auto& positions)
{
return [=](auto& nonLinOp, auto& fe, [[maybe_unused]] auto& req) { return checkCalculateAt(nonLinOp, fe,resultRequirements,expectedStress ,positions); };
}
```

There i think we have to capture the arguments by value instead of reference, since later these could by dangling. This is done by the lambda capture `[=]`